src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -102,6 +103,46 @@ public void setAttributes() {
         }
     }
 
+    /**
+     * Filters the attributes based on what attributes were chosen to be hidden.
+     *
+     * @param order the order that is unfiltered
+     * @return the order with attributes filtered where 4 represents a filtered attribute
+     */
+    private int[] filterAttributes(int[] order) {
+        boolean[] isHidden = new boolean[4];
+        GuiSettings currSettings = logic.getGuiSettings();
+        String currHiddenAttributes = currSettings.getHiddenAttributes().trim();
+
+        if (!currHiddenAttributes.equals("NONE")) {
+            String[] strArr = currHiddenAttributes.split(",");
+            try {
+                readHidden(strArr, isHidden);
```

Comment:
Same comment here regarding readHidden (should it be writeHidden instead)

Reply:
I have changed it to `setCurrentHiddenState()` to be consistent with the `setNewHiddenState()` method below.